### DISCUSSION TITLE
Current cloudflare bypass?

### DISCUSSION THREAD

**Discussion by yb-cs** [linweiyuan/go-chatgpt-api#263] [General]
Hello,

I was also looking into bypassing cloudflare. I was thinking exactly what you discovered in this blogpost (sending xmlhttp requests): https://linweiyuan.github.io/2023/03/14/%E4%B8%80%E7%A7%8D%E5%8F%96%E5%B7%A7%E7%9A%84%E6%96%B9%E5%BC%8F%E7%BB%95%E8%BF%87-Cloudflare-v2-%E9%AA%8C%E8%AF%81.html

But is there a more efficient way to do it without the browser have you discovered anything new or do you use anything new in this project?

Thank you for the feedback it is greatly appreciated

**Comment by linweiyuan** [comment]
Hi, this amazing repo is what you want: https://github.com/bogdanfinn/tls-client, and this is what I use since Apr (no browser, high performamce).

**Reply by yb-cs** [reply]
also found this is related also: 
**https://github.com/yifeikong/curl_cffi**

**Reply by yb-cs** [reply]
Oh thank you I will check it out ❤️ any specific configuration you used for it?

**Reply by linweiyuan** [reply]
it depends on the website you want to access.

no specific configurations currently.

which website you want to bypass?

**Reply by yb-cs** [reply]
Actually managed to bypass one request using the browser impersonation in curl_cffi but another endpoint has much more security and the browser impersonation is not enough. website is: visa.vfsglobal.com

Specific endpoint is: https://lift-api.vfsglobal.com/appointment/slots

Just trying to make a bot to check appointments and I dont want to use the browser.

**Reply by linweiyuan** [reply]
how to trogger this api, cannot find the entrypoint, always 403 if visit from browser directly, expected?

**Comment by yb-cs** [comment]
Yeah I am just trying to get slots using direct API call but seems tricky this is the landing page:

https://visa.vfsglobal.com/are/en/aut/login


I am basically trying to get all open appointments. I managed to do earliest date but that’s it.
